pkg/metrics/mocks_test.go

```diff
@@ -0,0 +1,89 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type mockPrometheusServer struct {
+	nodesForCluster map[string]int
+	labelSkip       map[string]struct{}
+	validTokens     []string
+}
+
+func newMockPrometheusServer() mockPrometheusServer {
+	return mockPrometheusServer{
+		nodesForCluster: map[string]int{},
+		labelSkip:       map[string]struct{}{},
+		validTokens:     []string{},
+	}
+}
+
+func (m *mockPrometheusServer) SetNodesForCluster(nodes int, cluster string, skipLabel bool) {
+	m.nodesForCluster[cluster] = nodes
+	if skipLabel {
+		m.labelSkip[cluster] = struct{}{}
+	}
+}
+
+func (m *mockPrometheusServer) Clear() {
+	m.nodesForCluster = map[string]int{}
+	m.validTokens = []string{}
+	m.labelSkip = map[string]struct{}{}
+}
+
+func (m *mockPrometheusServer) AddAuthToken(token string) {
+	m.validTokens = append(m.validTokens, token)
+}
+func (m *mockPrometheusServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// main serve method for the test http server
+	if !strings.HasSuffix(r.URL.String(), "/metrics") {
```

Comment:
Shouldn't the condition be reversed - if the URL does have the suffix `/metrics`?

Reply:
Yep - I also needed to include "/metrics" on the request. Changes have been made.